staging/src/k8s.io/apiserver/pkg/endpoints/request/webhook_duration.go

```diff
@@ -96,6 +115,19 @@ type LatencyTrackers struct {
 	// ValidatingWebhookTracker tracks the latency incurred in validating webhook(s).
 	// Validate webhooks are done in parallel, so max function is used.
 	ValidatingWebhookTracker DurationTracker
+
+	// StorageTracker tracks the latency incurred inside the storage layer.
```

Comment:
can you better define the boundaries on the storage layer?  it's the time it takes to send data to etcd and get the complete response or is it the time required to run admission, validation, and the etcd write?

Reply:
I don't understand how that is true since GuaranteedUpdate takes a function that determines the new object from the old object and then validates it 
 1. https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go#L528
 2. https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go#L540
 3. https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go#L633-L642
 4. https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/update.go#L218-L220
 5. https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/update.go#L171-L192

Reply:
hmm, right, I misinterpreted. I changed the approach, I am now tapping into the etcd3 `KV` client so now it measures etcd3 round trips only.